pkg/cluster/prepare_test.go

```diff
@@ -0,0 +1,219 @@
+package cluster_test
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/pctl/pkg/cluster"
+
+	gitfakes "github.com/weaveworks/pctl/pkg/git/fakes"
+)
+
+type mockTransport struct {
+	res *http.Response
+}
+
+func (t *mockTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return t.res, nil
+}
+
+var _ = Describe("prepare", func() {
+	When("dry run is set", func() {
+		It("can prepare the environment with everything that profiles needs without actually modifying the cluster", func() {
+			runner := &gitfakes.FakeRunner{}
+			content, err := ioutil.ReadFile(filepath.Join("testdata", "manifests.tar.gz"))
+			Expect(err).NotTo(HaveOccurred())
+			client := &http.Client{
+				Transport: &mockTransport{
+					res: &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       ioutil.NopCloser(bytes.NewReader(content)),
+					},
+				},
+			}
+			tmp, err := ioutil.TempDir("", "prepare_01")
+			Expect(err).NotTo(HaveOccurred())
+			p := &cluster.Preparer{
+				PrepConfig: cluster.PrepConfig{
+					Location: tmp,
+					DryRun:   true,
+				},
+				Fetcher: &cluster.Fetcher{
+					Client: client,
+				},
+				Applier: &cluster.Applier{
+					Runner: runner,
+				},
+			}
+			err = p.Prepare()
+			Expect(err).NotTo(HaveOccurred())
+			files, err := ioutil.ReadDir(tmp)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(files) > 0).To(BeTrue())
+			Expect(runner.RunCallCount()).To(Equal(1)) // there should be no call because of dryrun.
+			arg, args := runner.RunArgsForCall(0)
+			Expect(arg).To(Equal("kubectl"))
+			Expect(args).To(Equal([]string{"apply", "-f", tmp, "--dry-run=client"}))
```

Comment:
can we add a line to verify that the temp dir (Location) is cleaned up afterwards?